hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/format/cow/CopyOnWriteInputFormat.java

```diff
@@ -214,13 +213,7 @@ public FileInputSplit[] createInputSplits(int minNumSplits) throws IOException {
 
         // get the block locations and make sure they are in order with respect to their offset
         final BlockLocation[] blocks = fs.getFileBlockLocations(file, 0, len);
-        Arrays.sort(blocks, new Comparator<BlockLocation>() {
-          @Override
-          public int compare(BlockLocation o1, BlockLocation o2) {
-            long diff = o1.getLength() - o2.getOffset();
-            return Long.compare(diff, 0L);
-          }
-        });
+        Arrays.sort(blocks, CopyOnWriteInputFormat::compareBlockLocations);
 
```

Comment:
Does `Arrays.sort(blocks, Comparator.comparingLong(BlockLocation::getOffset));` work here ?

Reply:
Yes, should work fine here.

Reply:
Added the change

Reply:
@voonhous I don't see the change of using `comparingLong` in the latest commit.  Have you pushed your changes?

Reply:
My bad, i modified it to `comparing` instead of `comparingLong`. 

I have added the required changes. Apologies.

Reply:
No worries.  I'll merge this PR once CI passes.  Thanks for the fix!